### PR TITLE
Update CreateNewContextualCMutableTransaction for Sapling

### PR DESCRIPTION
**Pull request by str4d**
Part of #3063.

### CODE REVIEW

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
Pushed another commit updating the tests for the `CreateNewContextualCMutableTransaction` changes.

**Comment by str4d** [comment]
@zkbot r+

**Comment by str4d** [comment]
Hmm... The tests that failed on all the CI builders, worked fine for me locally.

**Comment by str4d** [comment]
Rebased on master, and they *still* pass locally. So it's not something that was recently introduced...

**Comment by str4d** [comment]
And I don't even understand why those tests would fail; nothing in this PR affects them.

@zkbot retry

**Comment by str4d** [comment]
Recreated the error on this branch (not merged to master) by running all gtests instead of just the affected ones. So there's some interdependency that has been introduced.

**Comment by mdr0id** [comment]
re-ACK

**Comment by str4d** [comment]
@zkbot r+

File: src/main.cpp [hunk 1]
```diff
@@ -6078,12 +6079,18 @@ CMutableTransaction CreateNewContextualCMutableTransaction(const Consensus::Para
...
-        // to set the expiry height to: min(activation_height - 1, default_expiry_height)
+        if (NetworkUpgradeActive(nHeight, consensusParams, Consensus::UPGRADE_SAPLING)) {
+            mtx.nVersionGroupId = SAPLING_VERSION_GROUP_ID;
+            mtx.nVersion = SAPLING_TX_VERSION;
+        } else {
+            mtx.nVersionGroupId = OVERWINTER_VERSION_GROUP_ID;
+            mtx.nVersion = OVERWINTER_TX_VERSION;
+            mtx.nExpiryHeight = std::min(
+                mtx.nExpiryHeight,
+                static_cast<uint32_t>(consensusParams.vUpgrades[Consensus::UPGRADE_SAPLING].nActivationHeight - 1));
```
Comment (bitcartel) [inline src/main.cpp]: Post merge review.  Nice to see a TODO comment predicting the future!

**Pull request merged**